Sources/Imperial/Routing/FederatedServiceRouter.swift

```diff
@@ -3,7 +3,7 @@ import Vapor
 
 /// Defines a type that implements the routing to get an access token from an OAuth provider.
 /// See implementations in the `Services/(Google|GitHub)/$0Router.swift` files
-public protocol FederatedServiceRouter {
+public protocol FederatedServiceRouter : class {
```

Comment:
This would require a major release and could potentially break existing user's code

Reply:
Yes, I can see that this could be a problem. I haven't been thinking much about Swift binary compatibility yet.